app/channel.go

```diff
@@ -596,10 +596,42 @@ func (a *App) postChannelPrivacyMessage(user *model.User, channel *model.Channel
 	return nil
 }
 
-func (a *App) RestoreChannel(channel *model.Channel) (*model.Channel, *model.AppError) {
+func (a *App) RestoreChannel(channel *model.Channel, userId string) (*model.Channel, *model.AppError) {
+	if channel.DeleteAt == 0 {
+		return nil, model.NewAppError("undeleteChannel", "api.channel.undelete_channel.undeleted.app_error", nil, "", http.StatusBadRequest)
+	}
+
 	if err := a.Srv.Store.Channel().Restore(channel.Id, model.GetMillis()); err != nil {
 		return nil, err
 	}
+	a.InvalidateCacheForChannel(channel)
+
+	var user *model.User
+	var err *model.AppError
+	user, err = a.Srv.Store.User().Get(userId)
+	if err != nil {
+		return nil, err
+	}
+
+	if user != nil {
+		T := utils.GetUserTranslations(user.Locale)
+
+		post := &model.Post{
+			ChannelId: channel.Id,
+			Message:   fmt.Sprintf(T("api.channel.undelete_channel.unarchived"), user.Username),
```

Comment:
The cleaner way would be to use the template feature from goi18n and not fmt. Something like
```suggestion
			Message:   T("api.channel.undelete_channel.unarchived", map[string]interface{}{"Username": user.Username}),
```

And change the message in `i18n/en.json` to `{{ .Username }}` unarchived the channel.`